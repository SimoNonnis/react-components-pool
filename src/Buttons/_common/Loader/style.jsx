import styled from 'styled-components';
import Spinner from '../../../Icons/Spinner';

export const SpinnerContainer = styled.div`
  position: relative;
`;

export const SpinnerStyled = styled(Spinner)`
  position: absolute;
  margin: 0 auto;
  left: 0;
  right: 0;
  top: -3px;
  ${({ size }) => size === 'large' && 'top: -5px'};
  ${({ size }) => size === 'small' && 'top: 0'};
`;
