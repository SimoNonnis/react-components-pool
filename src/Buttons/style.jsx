import React from 'react';
import styled, { css } from 'styled-components';
import { Link } from 'react-router-dom';
import {
  getColor,
  getColorOnHover,
  getBorderColor,
  getBorderColorOnHover,
  getBackgroundColor,
  getBackgroundColorOnHover,
  getPadding,
  getFontSize,
} from './utils';

const buttonStyle = css`
  -webkit-font-smoothing: antialiased;
  background-color: ${({ theme, styleType, type, disabled }) =>
    getBackgroundColor(theme, styleType, type, disabled)};
  border: 1px solid
    ${({ theme, styleType, type, disabled }) => getBorderColor(theme, styleType, type, disabled)};
  border-width: 1px;
  border-radius: ${({ rounded }) => `${rounded}px`};
  box-sizing: border-box;
  color: ${({ theme, styleType, type, disabled }) => getColor(theme, styleType, type, disabled)};
  cursor: ${({ disabled }) => (disabled ? 'no-drop' : 'pointer')};
  display: inline-block;
  font-family: ${({ theme }) => theme.base.fontFamily};
  font-size: ${({ size }) => getFontSize(size)};
  font-weight: 600;
  line-height: 24px;
  outline: 0;
  padding: ${({ size }) => getPadding(size)};
  text-align: center;
  text-shadow: none;
  white-space: nowrap;
  text-decoration: none;
  min-width: ${({ fullWidth, size }) => {
    if (fullWidth) return '100%';
    return size === 'small' ? '100px' : '140px';
  }};

  &:hover,
  &:focus {
    background-color: ${({ theme, styleType, type, disabled }) =>
      getBackgroundColorOnHover(theme, styleType, type, disabled)};
    border-color: ${({ theme, styleType, type, disabled }) =>
      getBorderColorOnHover(theme, styleType, type, disabled)};
    color: ${({ theme, styleType, type, disabled }) =>
      getColorOnHover(theme, styleType, type, disabled)};
    outline: 0;
  }
`;

const linkStyle = css`
  background-color: transparent;
  border: none;
  box-shadow: none;
  color: ${({ theme }) => theme.color.mediumBlue};
  text-decoration: underline;
  cursor: pointer;
  padding: 0;
  outline: 0;

  &:hover,
  &:focus {
    background-color: transparent;
    color: ${({ theme }) => theme.color.mediumBlue};
    text-decoration: none;
  }
`;

// eslint-disable-next-line
export const BtnLink = styled(({ external, fullWidth, to, theme, rounded, styleType, type, size, ...restProps}) => (external ? <a href={to} {...restProps} /> : <Link to={to} {...restProps} />))`
  ${buttonStyle};
`;
BtnLink.displayName = 'BtnLink';

export const Btn = styled.button`
  ${({ styleType, type }) => (styleType !== 'link' && type !== 'link' ? buttonStyle : linkStyle)};
`;
Btn.displayName = 'Btn';
