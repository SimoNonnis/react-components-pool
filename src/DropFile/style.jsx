import styled from 'styled-components';

import Dropzone from 'react-dropzone';

import UploadIcon from 'components/Icons/UploadIcon';
import Attention from 'components/Icons/Attention';
import P from 'components/Typography/P';
import Button from 'components/Buttons/Button';

export const StyledDropzone = styled(Dropzone)`
  border: 1px dashed ${({ theme }) => theme.color.mediumGrey};
  text-align: center;
  padding: ${({ theme }) => theme.spacing.lg} ${({ theme }) => theme.spacing.sm};
  cursor: pointer;
  background-color: ${({ theme }) => theme.color.white};
  width: 100%;
  box-sizing: border-box;
`;

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: ${({ theme }) => theme.color.veryDarkBlue};
  font-size: 16px;
  font-weight: 400;
`;

export const StyledUploadIcon = styled(UploadIcon)`
  color: ${({ theme }) => theme.color.lightGrey4};
`;

export const Label = styled(P)`
  margin-top: ${({ theme }) => theme.spacing.md};
`;

export const PickFileButton = styled(Button)`
  margin-top: ${({ theme }) => theme.spacing.sm};
`;

export const ContainerText = styled.div`
  padding: ${({ theme }) => theme.spacing.md};
  padding-bottom: 0;
  text-align: center;
`;

export const Message = styled(P)`
  color: ${({ error, theme }) => (error ? theme.color.red : theme.color.mediumGrey)};
  font-size: ${({ theme }) => theme.base.fontSize};
  font-style: ${({ error }) => (error ? 'normal' : 'italic')};
  margin-bottom: 0;
  margin-top: ${({ error, theme }) => (error ? theme.spacing.sm : '0')};
  ${({ error }) =>
    error &&
    `display: flex;
     align-items: center;
     justify-content: center;
  `};
`;

export const AttentionIcon = styled(Attention)`
  padding-right: ${({ theme }) => theme.spacing.xs};
  color: ${({ theme }) => theme.color.red};
`;
