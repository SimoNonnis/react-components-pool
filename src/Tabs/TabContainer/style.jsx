import styled from 'styled-components';
import { Flex } from 'grid-styled';

export const Tabs = styled(Flex)`
  font-size: 14px;
  border-bottom: 2px solid ${({ theme }) => theme.color.mediumBlue};
`;
export const Tab = styled.button`
  cursor: ${({ isActive }) => (isActive ? 'default' : 'pointer')};
  background-color: ${({ theme, isActive }) => (isActive ? theme.color.mediumBlue : '#e6e6e6')};
  color: ${({ theme, isActive }) => (isActive ? theme.color.white : 'inherit')};
  box-shadow: none;
  outline: 0;
  user-select: none;
  appearance: none;
  flex: 1 1 0;
  border: none;
  border-radius: 3px;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
  padding: 10px 0;
  font-size: inherit;
  margin-left: 3px;

  &:first-child {
    margin-left: 0;
  }

  > * {
    display: flex;
    align-items: center;
    justify-content: center;
  }
`;
