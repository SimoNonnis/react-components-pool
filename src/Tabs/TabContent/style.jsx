import styled from 'styled-components';

export const Container = styled.div`
  min-height: 200px;
  background-color: #fff;
  padding: 1.5em;
`;
