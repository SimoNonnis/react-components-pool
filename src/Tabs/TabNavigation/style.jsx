import styled from 'styled-components';
import Button from '../../Buttons/Button';

export const NavigationGroup = styled.div`
  background-color: #fff;
  padding: 1.5em;
  text-align: right;
`;

export const Text = styled.span`
  display: inline-block;
  vertical-align: middle;
  padding: 0 5px;
`;

export const Icon = styled.span`
  display: inline-block;
  vertical-align: middle;
`;

export const NavigationButton = styled(Button)`
  margin-left: 10px;
`;
