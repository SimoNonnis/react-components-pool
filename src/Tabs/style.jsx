import styled from 'styled-components';
import media from '../../utils/styles/mediaQueries';

export const LabelText = styled.span`
  display: none;
  ${media.tablet`
    display: inline-block;
    margin-left: .5em;
  `};
`;
