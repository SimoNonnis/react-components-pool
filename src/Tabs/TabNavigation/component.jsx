import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { ArrowLeft, ArrowRight } from '../../Icons';

import { NavigationGroup, Text, Icon, NavigationButton } from './style';
/**
 * TabNavigation renders navigation buttons for the tabs.
 * @param {function(number)} onChange
 * @param {number} currentIndex
 * @param {number} maxIndex
 * @returns {React.Component}
 */
class TabNavigation extends Component {
  constructor(props) {
    super(props);

    this.state = {
      currentIndex: this.props.currentIndex,
      maxIndex: this.props.maxIndex,
    };
  }

  componentWillReceiveProps({ currentIndex, maxIndex }) {
    this.setState({
      currentIndex,
      maxIndex,
    });
  }

  updateIndex = event => {
    event.preventDefault();
    const index = Number(event.currentTarget.value);
    const { onChange } = this.props;
    const { currentIndex } = this.state;

    this.setState({
      currentIndex: currentIndex + index,
    });

    onChange(currentIndex + index);
  };

  render() {
    const { currentIndex, maxIndex } = this.state;

    return (
      <NavigationGroup>
        {currentIndex !== 0 && (
          <NavigationButton type="button" value={-1} onClick={this.updateIndex}>
            <Icon>
              <ArrowLeft width={20} height={20} />
            </Icon>
            <Text>Previous</Text>
          </NavigationButton>
        )}
        {currentIndex !== maxIndex && (
          <NavigationButton
            type="button"
            disabled={currentIndex === maxIndex}
            value={1}
            onClick={this.updateIndex}
          >
            <Text>Next</Text>
            <Icon>
              <ArrowRight width={20} height={20} />
            </Icon>
          </NavigationButton>
        )}
      </NavigationGroup>
    );
  }
}

TabNavigation.propTypes = {
  currentIndex: PropTypes.number.isRequired,
  maxIndex: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default TabNavigation;
