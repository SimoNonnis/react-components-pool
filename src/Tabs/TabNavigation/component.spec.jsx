/* eslint-env jest */
import React from 'react';

import TabNavigation from './component';
import { NavigationButton } from './style';

describe('Test <TabNavigation />', () => {
  let wrapper;
  let callback;

  beforeEach(() => {
    callback = jest.fn();
    wrapper = shallow(<TabNavigation currentIndex={0} maxIndex={3} onChange={callback} />);
  });

  it('should find <TabNavigation /> component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should check if required prop currentIndex is present', () => {
    expect(wrapper.instance().props.currentIndex).toEqual(0);
  });

  it('should check if required prop maxIndex is present', () => {
    expect(wrapper.instance().props.maxIndex).toEqual(3);
  });

  it('should check if required prop handleNavigationClick() is present', () => {
    expect(wrapper.instance().props.onChange).toEqual(callback);
  });

  it('should render 1 button on initial mount', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render multiple buttons when not on the beginning or end tab', () => {
    wrapper.setState({ currentIndex: 2 });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render 1 button on the last tab', () => {
    wrapper.setState({ currentIndex: 3 });
    expect(wrapper).toMatchSnapshot();
  });

  it('should check if prev button is enabled after clicking on the next button', () => {
    const button = wrapper.find(NavigationButton);
    button.simulate('click', {
      preventDefault() {},
      currentTarget: {
        value: 1,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should check if next button is disabled when last tab is showing', () => {
    wrapper.setState({ currentIndex: 3 });

    expect(wrapper).toMatchSnapshot();
  });
});
