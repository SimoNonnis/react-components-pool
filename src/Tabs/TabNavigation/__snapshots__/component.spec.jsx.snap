// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <TabNavigation /> should check if next button is disabled when last tab is showing 1`] = `
<styled.div>
  <Styled(Button)
    onClick={[Function]}
    type="button"
    value={-1}
  >
    <styled.span>
      <ArrowLeft
        className="arrowLeftIcon"
        classNamePath="arrowLeftIconPath"
        fill="currentColor"
        height={20}
        width={20}
      />
    </styled.span>
    <styled.span>
      Previous
    </styled.span>
  </Styled(Button)>
</styled.div>
`;

exports[`Test <TabNavigation /> should check if prev button is enabled after clicking on the next button 1`] = `
<styled.div>
  <Styled(Button)
    onClick={[Function]}
    type="button"
    value={-1}
  >
    <styled.span>
      <ArrowLeft
        className="arrowLeftIcon"
        classNamePath="arrowLeftIconPath"
        fill="currentColor"
        height={20}
        width={20}
      />
    </styled.span>
    <styled.span>
      Previous
    </styled.span>
  </Styled(Button)>
  <Styled(Button)
    disabled={false}
    onClick={[Function]}
    type="button"
    value={1}
  >
    <styled.span>
      Next
    </styled.span>
    <styled.span>
      <ArrowRight
        className="arrowRightIcon"
        classNamePath="arrowRightIconPath"
        fill="currentColor"
        height={20}
        width={20}
      />
    </styled.span>
  </Styled(Button)>
</styled.div>
`;

exports[`Test <TabNavigation /> should find <TabNavigation /> component 1`] = `
<styled.div>
  <Styled(Button)
    disabled={false}
    onClick={[Function]}
    type="button"
    value={1}
  >
    <styled.span>
      Next
    </styled.span>
    <styled.span>
      <ArrowRight
        className="arrowRightIcon"
        classNamePath="arrowRightIconPath"
        fill="currentColor"
        height={20}
        width={20}
      />
    </styled.span>
  </Styled(Button)>
</styled.div>
`;

exports[`Test <TabNavigation /> should render 1 button on initial mount 1`] = `
<styled.div>
  <Styled(Button)
    disabled={false}
    onClick={[Function]}
    type="button"
    value={1}
  >
    <styled.span>
      Next
    </styled.span>
    <styled.span>
      <ArrowRight
        className="arrowRightIcon"
        classNamePath="arrowRightIconPath"
        fill="currentColor"
        height={20}
        width={20}
      />
    </styled.span>
  </Styled(Button)>
</styled.div>
`;

exports[`Test <TabNavigation /> should render 1 button on the last tab 1`] = `
<styled.div>
  <Styled(Button)
    onClick={[Function]}
    type="button"
    value={-1}
  >
    <styled.span>
      <ArrowLeft
        className="arrowLeftIcon"
        classNamePath="arrowLeftIconPath"
        fill="currentColor"
        height={20}
        width={20}
      />
    </styled.span>
    <styled.span>
      Previous
    </styled.span>
  </Styled(Button)>
</styled.div>
`;

exports[`Test <TabNavigation /> should render multiple buttons when not on the beginning or end tab 1`] = `
<styled.div>
  <Styled(Button)
    onClick={[Function]}
    type="button"
    value={-1}
  >
    <styled.span>
      <ArrowLeft
        className="arrowLeftIcon"
        classNamePath="arrowLeftIconPath"
        fill="currentColor"
        height={20}
        width={20}
      />
    </styled.span>
    <styled.span>
      Previous
    </styled.span>
  </Styled(Button)>
  <Styled(Button)
    disabled={false}
    onClick={[Function]}
    type="button"
    value={1}
  >
    <styled.span>
      Next
    </styled.span>
    <styled.span>
      <ArrowRight
        className="arrowRightIcon"
        classNamePath="arrowRightIconPath"
        fill="currentColor"
        height={20}
        width={20}
      />
    </styled.span>
  </Styled(Button)>
</styled.div>
`;
